packages/react/hover-card/src/HoverCard.tsx

```diff
@@ -0,0 +1,289 @@
+      data-state={context.open ? 'open' : 'closed'}
+      {...triggerProps}
+      as={as}
+      ref={composedTriggerRef}
+      onMouseEnter={composeEventHandlers(props.onMouseEnter, context.onMouseEnter)}
+      onMouseLeave={composeEventHandlers(props.onMouseLeave, context.onMouseLeave)}
+    />
+  );
+
+  return context.hasCustomAnchor ? (
+    trigger
+  ) : (
+    <PopperPrimitive.Anchor as={Slot}>{trigger}</PopperPrimitive.Anchor>
+  );
+}) as HoverCardTriggerPrimitive;
+
+HoverCardTrigger.displayName = TRIGGER_NAME;
+
+/* -------------------------------------------------------------------------------------------------
+ * HoverCardContent
+ * -----------------------------------------------------------------------------------------------*/
+
+const CONTENT_NAME = 'HoverCardContent';
+
+type HoverCardContentOwnProps = Polymorphic.Merge<
+  Polymorphic.OwnProps<typeof HoverCardContentImpl>,
+  {
+    /**
+     * Used to force mounting when more control is needed. Useful when
+     * controlling animation with React animation libraries.
+     */
+    forceMount?: true;
+  }
+>;
+
+type HoverCardContentPrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof HoverCardContentImpl>,
+  HoverCardContentOwnProps
+>;
+
+const HoverCardContent = React.forwardRef((props, forwardedRef) => {
+  const { forceMount, ...contentProps } = props;
+  const context = useHoverCardContext(CONTENT_NAME);
+
+  return (
+    <Presence present={forceMount || context.open}>
+      <HoverCardContentImpl
+        data-state={context.open ? 'open' : 'closed'}
+        {...contentProps}
+        onMouseEnter={composeEventHandlers(props.onMouseEnter, context.onMouseEnter)}
+        onMouseLeave={composeEventHandlers(props.onMouseLeave, context.onMouseLeave)}
+        ref={forwardedRef}
+      />
+    </Presence>
+  );
+}) as HoverCardContentPrimitive;
+
+type PopperPrimitiveOwnProps = Polymorphic.OwnProps<typeof PopperPrimitive.Content>;
+type HoverCardContentImplOwnProps = Polymorphic.Merge<
+  PopperPrimitiveOwnProps,
+  {
+    /**
+     * Whether the `HoverCard` should render in a `Portal`
+     * (default: `true`)
+     */
+    portalled?: boolean;
+  }
+>;
+
+type HoverCardContentImplPrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof PopperPrimitive.Content>,
+  HoverCardContentImplOwnProps
+>;
+
+const HoverCardContentImpl = React.forwardRef((props, forwardedRef) => {
+  const { portalled = true, ...contentProps } = props;
+  const context = useHoverCardContext(CONTENT_NAME);
+
+  const PortalWrapper = portalled ? Portal : React.Fragment;
+
+  return (
+    <PortalWrapper>
+      <PopperPrimitive.Content
+        id={context.contentId}
+        {...contentProps}
+        ref={forwardedRef}
+        style={{
+          ...contentProps.style,
+          // re-namespace exposed content custom property
+          ['--radix-hovercard-content-transform-origin' as any]: 'var(--radix-popper-transform-origin)',
```

Comment:
I believe the casing would be

```suggestion
          ['--radix-hover-card-content-transform-origin' as any]: 'var(--radix-popper-transform-origin)',
```

?